### ISSUE TITLE
could not launch process: EOF

### ISSUE THREAD

**Issue by warjiang** [derekparker/delve#1165]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.0.0
Build: db435c184eb79976087fc0cb36c2867c737a9066
```
2. What version of Go are you using? (`go version`)?
```
go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/dingwenjiang/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/dingwenjiang/go"
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/ly/7p11f6gn5v71sspkdwx4qhgc0000gn/T/go-build573955664=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
3. What operating system and processor architecture are you using?
```
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.4
BuildVersion:	17E199


uname -a
Darwin dingwenjiangdeMacBook-Pro.local 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
```
4. What did you do?
```
go get -u github.com/derekparker/delve/cmd/dlv
go build -gcflags "-N -l" -a github.com/golang/example/hello
$GOPATH/bin/dlv exec hello
could not launch process: EOF
```
5. What did you expect to see?
```
To dlv start a debug session
```
6. What did you see instead?
```
could not launch process: EOF
```

**Comment by luq885** [comment]
Same issue and same MAC OS version, just updated today.

**Comment by mervinkid** [comment]
Same as @luq885 .

**Comment by TomorrowWu** [comment]
me too. @luq885

**Comment by TomorrowWu** [comment]
my go version is 1.10.1

**Comment by TomorrowWu** [comment]
IDEA version is 2018.1

**Comment by Khamliuk** [comment]
Same issue as @warjiang  and now I updated Mac OS to 10.13.4

**Comment by warjiang** [comment]
@Khamliuk everything is ok when i used Mac OS 10.13.3。After i update Mac OS to 10.13.4, I can't use delve to debug programe. It seems the problem of codesign on MacOS

**Comment by Khamliuk** [comment]
@warjiang Yes, I have the same situation

**Comment by kcmannem** [comment]
same, I believe it was the 10.13.4 update that caused this. I'm in this situation after updating.

**Comment by antuspenskiy** [comment]
Im on macOS High Sierra 10.13.3 and have same problem!

**Comment by TomorrowWu** [comment]
Did you upgrade the Command Line Tool Tools? @antuspenskiy

**Comment by TomorrowWu** [comment]
I upgrade macOS High Sierra 10.13.4, Go 1.10.1  in the same day

**Comment by usualwyy** [comment]
I feel soooo regret for my upgradtion to 10.13.4.
I should not do this dumb thing.

**Comment by dgorohov** [comment]
Same symptoms after upgrading to 10.13.4 ;( Please helppppp

**Comment by mente** [comment]
For me the problem started from Command Line Tools update, didn't make update to 10.13.4 yet. So the problem is somewhere in command line tools :-/

**Comment by TomorrowWu** [comment]
command line tools

**Comment by dareniott** [comment]
Idk what was updated.  I was debugging this morning, after lunch failed to launch debugger.  This is a major issue for me, can't debug anything with GoLand.

**Comment by usualwyy** [comment]
In my case,another issue came with this one at the same time:
Xcode stop response during debug the pure C++ project.
Anyone got the same issue?

**Comment by antuspenskiy** [comment]
> Did you upgrade the Command Line Tool Tools? @antuspenskiy

Yes it was updated during MacOS 10.13.3-10.13.4.

**Comment by TomorrowWu** [comment]
I rollback my CommandLineTools,
Temporarily solved the problem by removing new Command Line Tools
$ sudo rm -rf /Library/Developer/CommandLineTools

then go to page https://developer.apple.com/download/more/
download and install previous version

Command Line Tools (macOS 10.13) for Xcode 9.2 - Dec 4, 2017

**Comment by midona-rhel** [comment]
Experiencing the exact same issue, rollback of CLT worked however. Specifically it seems like no debug file is created.

**Comment by warjiang** [comment]
@TomorrowWu thanks, after i rollback command line tools, delve can work

**Comment by UrNas** [comment]
@TomorrowWu thank you

**Comment by aarzilli** [comment]
Somebody having this problem should try adding `conn.gcmdok = false` after `$GOPATH/src/github.com/derekparker/delve/pkg/proc/gdbserial/gdbserver_conn.go:109`. If it doesn't work, don't post "it doesn't work", turn on logGdbWire and post the output.

**Comment by savanu** [comment]
So I downloaded the latest master and ran a debug session and it worked fine. But the version from brew does not work. 

go version go1.10.1 darwin/amd64

sw_vers
ProductName:    Mac OS X
ProductVersion: 10.13.4
BuildVersion:   17E199

go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/merlin/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/merlin/gopath"
GORACE=""
GOROOT="/usr/local/opt/go/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/opt/go/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/5j/r39qwlt9257g6fgwrgh6k_km0000gn/T/go-build629340659=/tmp/go-build -gno-record-gcc-switches -fno-common"

Working:
Delve Debugger
Version: 1.0.0
Build: db435c184eb79976087fc0cb36c2867c737a9066

Broken for this scenario:
Delve Debugger
Version: 1.0.0
Build: 678576d2ce3b14f7b6b59692669b8acbd7c427ff

**Comment by MarkSRobinson** [comment]
I have a workaround.

`sudo rm -rf /Library/Developer/CommandLineTools. `
 
Go to https://developer.apple.com/download/more/ and download `Command Line Tools (macos 10.13) for XCode 9.2.`

This will restore debugging until the issue gets resolved upstream.

**Comment by Khamliuk** [comment]
@MarkSRobinson I can't find this version of Command Line Tools on the site

**Comment by YaoJunZhao** [comment]
@Khamliuk https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_macOS_10.13_for_Xcode_9.2/Command_Line_Tools_macOS_10.13_for_Xcode_9.2.dmg

**Comment by rcoreilly** [comment]
@aarzilli I couldn't follow those instructions for patching gdbserver.go but this seems to work:

```go
diff --git a/pkg/proc/gdbserial/gdbserver.go b/pkg/proc/gdbserial/gdbserver.go
index a201e57..4791f98 100644
--- a/pkg/proc/gdbserial/gdbserver.go
+++ b/pkg/proc/gdbserial/gdbserver.go
@@ -418,6 +418,9 @@ func LLDBLaunch(cmd []string, wd string) (*Process, error) {
 
        p := New(proc.Process)
        p.conn.isDebugserver = isDebugserver
+       if isDebugserver {
+               p.gcmdok = false
+       }
 
        if listener != nil {
                err = p.Listen(listener, cmd[0], 0)
```

**Comment by aarzilli** [comment]
@rcoreilly: cool. Can you try PR #1169, I'd like to disable this only for the specific version of debugserver that is affect.

**Comment by rcoreilly** [comment]
yep that works!  I didn't see that I could click on that d3ad0cf link on your comment above and get that patch..

**Comment by aarzilli** [comment]
@rcoreilly Thank you.

**Comment by jicki** [comment]
@TomorrowWu  thank you ~

**Comment by dkorhov** [comment]
still do not work with applied [comment](https://github.com/derekparker/delve/issues/1165#issuecomment-378002963) and d3ad0cf

**Comment by Lqlsoftware** [comment]
@dkorhov try his method @TomorrowWu. It's new version of Command Line Tools which lead to this issue.

**Comment by suiwenfeng** [comment]
@YaoJunZhao Thanks

**Comment by RoarkeRandall** [comment]
Weird, I down graded the CLI tools like @YaoJunZhao suggested and it worked, but this morning its no longer working. I wonder if macOS automatically upgraded the cli tools again?

**Comment by levrik** [comment]
@RoarkeRandall could be. My app store wants to upgrade it again but I just declined the terms and conditions when they showed up. Now it waits for manual approve.

**Comment by wubingyang** [comment]
@姚俊兆   我没有苹果的开发者账号没办法下载那个工具.. 能否给我邮箱发一份谢谢120549547@qq.com

**Comment by JCzz** [comment]
I also ran into this problem today - any work regarding this on this great debugger.

**Comment by livio-a** [comment]
@RoarkeRandall it did, at least for me:
![image](https://user-images.githubusercontent.com/9405495/38405401-f70b7e0a-396f-11e8-9041-3a7aab3b1849.png)
I disabled auto updates for now - waiting for a proper solution

**Comment by MvHugo** [comment]
Hi, Aarzilli,

Manually merged your fix and seems did not fix. So have to use @TomorrowWu's workaround that roll back the command line tools.

**Comment by olostan** [comment]
`p.gcmdok = false` worked for 902.0.39.1 (patched, re-builded glv binary)

**Comment by 0915240** [comment]
goland track:
https://youtrack.jetbrains.com/issue/GO-5527

**Comment by dlsniper** [comment]
@adamgoose are you experiencing the same issue after the patch or not?

**Comment by aarzilli** [comment]
@MvHugo 
@rcoreilly 
if the patch doesn't work you need to enable logGdbWire and post the output, or nothing will ever be fixed.

**Comment by rcoreilly** [comment]
all versions of the patch have worked for me, on MacOs 10.13.4, using xcode / command line tools 9.3, installed Mar 30.  perhaps also for those whom it is not working, they could add some fmt.Printf's  in the patch code to see if g.gcmdok = false is being run, and what the version string is reporting?

**Comment by aarzilli** [comment]
Enabling logGdbWire will print the version string, among other things.

**Comment by pscoumar** [comment]
@TomorrowWu You saved me... Thanks...

**Comment by tfnaylor** [comment]
Any estimate as to when this patch might be merged into master so that the rest of us can get it through a simple go get?

**Comment by pnpavlov** [comment]
@rcoreilly thank you for the workaround. worked well for me with latest command line tools as of today.

**Comment by banks** [comment]
@rcoreilly patch works here for me on latest everything. Thanks!

**Comment by sourcec0de** [comment]
For anyone waiting for this patch, you can patch manually like this.
Run these instructions inside of the 
```
git remote add aarzilli git@github.com:aarzilli/delve.git
git pull aarzilli debugserverfix
GO15VENDOREXPERIMENT=1 CERT=dlv-cert make install
```

If you need to reload your shell `exec $SHELL`

**Comment by okabbas** [comment]
Is there a permanent solution to this problem? Is this bug fixed?

**Comment by dlsniper** [comment]
As I know many of the GoLand users are watching this issue, GoLand 2018.1.1 / 2017.3.4 were rolled-out with the patch from #1169. Thank you to the Delve team for the fix on this.

**Comment by aarzilli** [comment]
@okabbas as far as I know the bug still exists and will resurface when Apple releases a new version of debugserver: https://bugs.llvm.org/show_bug.cgi?id=36968

**Comment by dlsniper** [comment]
@derekparker / @aarzilli do you guys think it's possible to tag a new release to include this patch in it? There are a few other commits on master that are not in 1.0.0, but this one would be worthy of at least of 1.0.1 release. Thank you!

**Comment by radugh** [comment]
`go get -u github.com/derekparker/delve/cmd/dlv` will solve this issue.

**Comment by jpap** [comment]
For those using Homebrew like me, upgrade to the `HEAD` version:
```
brew upgrade delve --HEAD
```
Works on 10.13.4 with latest Command Line Tools 9.3.

**Comment by henryse** [comment]
Here is how I fixed it for GoLang:

1. Build the latest version of dlv: go get -u github.com/derekparker/delve/cmd/dlv
2. Now find your "go/bin" directory, you should see the newly built dlv located in the directory
3. Rename GoLang's old dlv to dlv.old, my GoLang's dlv was located in: /Applications/GoLand.app/Contents/plugins/go/lib/dlv/mac/dlv
4. Copy "go/bin/dlv" to /Applications/GoLand.app/Contents/plugins/go/lib/dlv/mac/dlv

**Comment by zolotov** [comment]
@henryse, I’d appreciate if yo remove the instruction. 

1. 2018.1.1 and 2017.3.5 with the fix were released more than a week ago
2. The instruction of harmful since it breaks signature of Goland application and breaks future patch update

**Comment by charnlsxy** [comment]
maybe someone got same situation as me?
moc os
goland 
could not launch process: EOF 
exit Goland
move /Volumes/GoLand to application
push-off   /Volumes/GoLand
restart
works...

**Comment by zolotov** [comment]
@charnlsxy what's the IDE version you use? Could you please file an issue at https://youtrack.jetbrains.com and attach logs there (Help | Collect logs and show in finder)?

**Comment by charnlsxy** [comment]
@zolotov version2018.1
i don't know why
![8c6f8156-533b-439f-83e0-9ad53be22b0b](https://user-images.githubusercontent.com/13843868/39362711-fd665f2c-4a59-11e8-990c-bcbe63dd7653.png)

**Comment by charnlsxy** [comment]
@zolotov i don't know why
![7f17556f-7e7c-4a4f-b98f-656d8e4a087b](https://user-images.githubusercontent.com/13843868/39362799-4cca82a0-4a5a-11e8-8ad8-15e35ff387d8.png)

**Comment by edudojevic** [comment]
Update to 2018.1.3 and the Go plugin fixes the issue for me.

**Comment by willis7** [comment]
I'm getting this with VS Code. I tried the tip to remove latest XCode - no joy :-(

**Comment by dotw** [comment]
I still have the same issue with the latest codes. go get -u github.com/derekparker/delve/cmd/dlv doesn't work for me.

**Comment by VitaliiLakusta** [comment]
What worked for me (assuming you use GoLand IDE): 
1. Going to Delve settings in Go. Press cmd+shift+a, type "delve", hit enter.
2. Select box "Use native backend"

**Comment by levrik** [comment]
@willis7 Yes. The VS Code extension installs `delve` via `go get` on first start.
There's also a command in VS Code by the extension to update all it's Go dependencies.

**Comment by fiskeben** [comment]
@dotw Make sure you:

* haven't installed `dlv` in another way (for example with `brew`). Verify with `which dlv` that you're actually running the version installed by `go get`.
* have `$GOPATH/bin` (or something equivalent) in your `$PATH`.

That was the case for me and after fixing that it seems to work with the version installed by `go get`.

**Comment by andofcourse** [comment]
Has there been a clear solution to this? I'm still running into the same error and I just did `go get` on delve today (so its the most recent version), but I'm still getting `could not launch process: stub exited while waiting for connection: exit status 0`

**Comment by jrc60752** [comment]
@radugh 's  suggestion looked by far the easiest to give a shot and it worked.

I didn't touch XCode or any OS tools just ran:
go get -u github.com/derekparker/delve/cmd/dlv


OS: 10.13.4

$ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
package-id: com.apple.pkg.CLTools_Executables
version: 9.3.0.0.1.1521514116
volume: /
location: /
install-time: 1525178199
groups: com.apple.FindSystemFiles.pkg-group

$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**Comment by dlsniper** [comment]
@WangChangQin Please move this thread to our tracker at https://youtrack.jetbrains.com/issues/Go but this problem should have been solved in 2018.1.1, a long time ago.

**Comment by sjdweb** [comment]
As @levrik suggests, for VS Code you can update all of your tools via the Command Pallette.
This is accessed via `CMD+Shift+P` on MacOS, search for `Go: Install/Update Tools` and tick at least `dlv (Debugging)` to update delve.
